Johnny-jbuccola/homework_3_JSON/json-hw.js

```diff
@@ -0,0 +1,42 @@
+
+/*
+1. Convert the string `jsonString` to a Javascript Object and store it in a variable
+called `flickerObj`
+*/
+var flickerObj = JSON.parse(jsonString)
+
+
+
+/*
+2. Iterate (this means 'use a loop') and do a console.log of each of the item titles
+with the date the photo was taken. The first console.log should return the following:
+
+"Edelweiss Air, Airbus A320-214, HB-IJU 2016-02-23T18:48:14Z"
+
+*/
+for (i=0; i < flickerObj.items.length ; i++){
+    console.log(flickerObj.items[i].title + ", " + flickerObj.items[i].date_taken);
+}
+
+/*
+3. Create a new array called 'links'. Iterate through the items array in flickerObj and store the the links to each item in the 'links' array.
+*/
+
+var links = []
+
+for (i=0; i < flickerObj.items.length ; i++) {
+    links.push(flickerObj.items[i].link)
+}
+
+/*
+4. Convert the 'links' array into a JSON string. Do a console.log of that JSON string, copy the results from the terminal and paste them into the input box at http://jsonlint.com/
+Did it create a valid JSON string?
+*/ 
+console.log(JSON.stringify(links))
+
+//JBUCCOLA NOTE: Why this doesn't work? from the documentation on JSON.stringify(), it seems like the entire thing should be enclosed in ''? It fails in the JSON linter.
```

Comment:
Although the printout looks like it's an array when returned in terminal, take a look at the `typeof`:

     $ console.log(typeof JSON.stringify(links));
     > string